infra-as-code/bicep/machinelearning.bicep

```diff
@@ -442,16 +348,18 @@ resource endpointDiagnostics 'Microsoft.Insights/diagnosticSettings@2021-05-01-p
   }
 }
 
+// Production readiness change: Client applications that run from compute on Azure should use managed identities instead of
+// pre-shared keys. This sample implementation uses a pre-shared key, and should be rewritten to use the managed identity
+// provided by Azure Web Apps.
 @description('Key Vault Secret: The Managed Online Endpoint key to be referenced from the Chat UI app.')
 resource managedEndpointPrimaryKeyEntry 'Microsoft.KeyVault/vaults/secrets@2023-07-01' = {
   parent: keyVault
   name: 'chatApiKey'
   properties: {
-    value: machineLearning::onlineEndpoint.listKeys().primaryKey
+    value: chatProject::endpoint.listKeys().primaryKey // This key is technically already in Key Vault, but it's name is not something that is easy to reference.
     contentType: 'text/plain'
     attributes: {
       enabled: true

```

Comment:
Go ahead and set an expiry date on this `secret` of whatever the default policy is for ALZ secrets.  I know it doesn't change anything about the policy being applied, but it does show good intent.  This is something I'll backport to Baseline (and Basic) as well.

Reply:
Lets defer this for now and address in all up policy refactoring work we will kick off post this, don't want to introduce a change above and beyond what has been tested end to end and is ready for this release